Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm

```diff
@@ -705,11 +711,32 @@ - (void)loadView
 #endif
 
 #if ENABLE(LINEAR_MEDIA_PLAYER)
-        _enterVideoFullscreenButton = [self _createButtonWithExtrinsicContentSize:buttonSize];
-        [_enterVideoFullscreenButton setConfiguration:cancelButtonConfiguration];
-        [_enterVideoFullscreenButton setImage:[[UIImage systemImageNamed:@"video"] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] forState:UIControlStateNormal];
-        [_enterVideoFullscreenButton sizeToFit];
+        // calculate title width requirement.
+        auto fullscreenString = @"View Spatial";
+        auto fullscreenTitleFont = [UIFont systemFontOfSize:17 weight:UIFontWeightMedium];
+        auto buttonTitle = adoptNS([[NSMutableAttributedString alloc] initWithString:fullscreenString]);
+        [buttonTitle addAttribute:NSFontAttributeName value:fullscreenTitleFont range:NSMakeRange(0, fullscreenString.length)];
+        size_t textWidth = [buttonTitle size].width;
+
+        // calculate spatial icon width requirement.
+        auto spatialImage = [UIImage systemImageNamed:@"cube" withConfiguration:[UIImageSymbolConfiguration configurationWithPointSize:17 weight:UIImageSymbolWeightMedium]];
+        size_t imageWidth = [spatialImage size].width;
+
+        _enterVideoFullscreenButton = [self _createButtonWithExtrinsicContentSize:CGSizeMake(18 + imageWidth + 9 + textWidth + 22, buttonSize.height)];
+        UIButtonConfiguration *fullscreenButtonConfiguration = [UIButtonConfiguration filledButtonConfiguration];
+        fullscreenButtonConfiguration.imagePadding = 9;
+        fullscreenButtonConfiguration.contentInsets = NSDirectionalEdgeInsetsMake(12, 18, 12, 22);
+        fullscreenButtonConfiguration.titleLineBreakMode = NSLineBreakByClipping;
+        [_enterVideoFullscreenButton setConfiguration:fullscreenButtonConfiguration];
+        [_enterVideoFullscreenButton setContentHorizontalAlignment:UIControlContentHorizontalAlignmentLeft];
+
+        [_enterVideoFullscreenButton setImage:[spatialImage imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] forState:UIControlStateNormal];
+        [_enterVideoFullscreenButton setAttributedTitle:buttonTitle.get() forState:UIControlStateNormal];
```

Comment:
I think you should be able to set `image` and `attributedTitle` on the `UIButtonConfiguration`?

Reply:
done